tools_docker/test_docker/fixtures.py

```diff
@@ -0,0 +1,124 @@
+import os
+from distutils.util import strtobool
+from time import sleep
+
+from dotenv import dotenv_values
+import pytest
+import subprocess
+import testinfra
+
+
+test_config_value = dotenv_values(os.path.dirname(os.path.realpath(__file__))+'/test_config.env')
+STOP_DOCKER_AT_END = strtobool(test_config_value['STOP_DOCKER_AT_END'])
+DOCKER_TRACIM_IMAGE = test_config_value['DOCKER_TRACIM_IMAGE']
+DOCKER_ELASTICSEARCH_IMAGE =  test_config_value['DOCKER_ELASTICSEARCH_IMAGE']
+DOCKER_COLLABORA_IMAGE = test_config_value['DOCKER_COLLABORA_IMAGE']
+DOCKER_TEST_NETWORK = test_config_value['DOCKER_TEST_NETWORK']
+SLEEP_AFTER_START_TRACIM_DOCKER = int(test_config_value['SLEEP_AFTER_START_TRACIM_DOCKER'])
+SLEEP_AFTER_START_ELASTICSEARCH_DOCKER = int(test_config_value['SLEEP_AFTER_START_ELASTICSEARCH_DOCKER'])
+SLEEP_AFTER_START_COLLABORA_DOCKER = int(test_config_value['SLEEP_AFTER_START_COLLABORA_DOCKER'])
+TRACIM_DOMAIN_NAME = test_config_value['TRACIM_DOMAIN_NAME']
+TRACIM_ETC_FOLDER = test_config_value['TRACIM_ETC_FOLDER']
+TRACIM_VAR_FOLDER = test_config_value['TRACIM_VAR_FOLDER']
+
+@pytest.fixture(scope='session')
+def tracim_env_var_file_path():
+    return os.path.dirname(os.path.realpath(__file__))+'/tracim.env'
+
+@pytest.fixture(scope='session')
+def tracim_env_var_params(tracim_env_var_file_path):
+    env_vars=dotenv_values(tracim_env_var_file_path)
+    return env_vars
+
+
+@pytest.fixture(scope='session')
+def tracim_docker_params(tracim_env_var_params):
+    params = [
+        'docker',
+        'run',
+        '--name', 'tracim.test',
+        '--hostname', "tracim.test",
+        '-d',
+        '--network',
+        DOCKER_TEST_NETWORK,
+    ]
+    for name,value in tracim_env_var_params.items():
+        params.append('-e')
+        params.append('{}={}'.format(name, value))
+    if TRACIM_ETC_FOLDER:
+        params.append('-v')
+        params.append('{}:/etc/tracim'.format(TRACIM_ETC_FOLDER))
+    if TRACIM_VAR_FOLDER:
+        params.append('-v')
+        params.append('{}:/var/tracim'.format(TRACIM_VAR_FOLDER))
+    params.append(DOCKER_TRACIM_IMAGE)
+    return params
+
+@pytest.fixture(scope='session')
+def client():
+    return testinfra.get_host('local://')
```

Comment:
Should the host be configurable? Probably not, but just checking out of precaution.

Reply:
... currently we do not need to as this is only for test.
I do think we should later find a way to split:
- all the docker running code, used here to setting a working env for test.
- the test itselfs.

Idea is that the tests itself (in another .py file) make sense to be runned outside of this scope (Tracim on production environment for example !) but not the current state of fixture with all the config, docker starting, etc... mecanism.
I'm not sure how to this now, and we currently does not need this feature, but this is probably not so complex to do (just need to find a way to override properly tracim fixture).